nipype/interfaces/dtitk/registration.py

```diff
@@ -1,394 +1,430 @@
+                               xor=['transform'])
+    euler = traits.Tuple((traits.Float(), traits.Float(), traits.Float()),
+                         desc='(theta, phi, psi) in degrees',
+                         xor=['transform'], argstr='-euler %g %g %g')
+    deformation = traits.Tuple((traits.Float(),) * 6,
+                               desc='(xx,yy,zz,xy,yz,xz)', xor=['transform'],
+                               argstr='-deformation %g %g %g %g %g %g')
+
+
+class AffScalarVolOutputSpec(TraitedSpec):
+    out_file = File(desc='moved volume', exists=True)
+
+
+class AffScalarVol(CommandLineDtitk):
     """
     Applies affine transform to a scalar volume
 
-                Example
-                -------
-
-                >>> import nipype.interfaces.dtitk as dtitk
-                >>> node = dtitk.affScalarVolTask()
-                >>> node.inputs.in_volume = 'fa.nii.gz'
-                >>> node.inputs.in_xfm = 'ants_Affine.txt'
-                >>> node.run() # doctest: +SKIP
-                """
-    input_spec = affScalarVolInputSpec
-    output_spec = affScalarVolOutputSpec
+    Example
+    -------
+
+    >>> from nipype.interfaces import dtitk
+    >>> node = dtitk.AffScalarVol()
+    >>> node.inputs.in_file = 'im1.nii'
+    >>> node.inputs.transform = 'im_affine.aff'
+    >>> node.cmdline
+    'affineScalarVolume -in im1.nii -interp 0 -out im1_affxfmd.nii -trans
+     im_affine.aff'
+    >>> node.run() # doctest: +SKIP
+    """
+    input_spec = AffScalarVolInputSpec
+    output_spec = AffScalarVolOutputSpec
     _cmd = 'affineScalarVolume'
 
-    def _list_outputs(self):
-        outputs = self.output_spec().get()
-        outputs['out_file'] = os.path.abspath(self.inputs.out_file)
-        return outputs
+    def _format_arg(self, name, spec, value):
+        if name == 'interpolation':
+            value = {'trilinear': 0, 'NN': 1}[value]
+        return super(AffScalarVol, self)._format_arg(name, spec, value)
+
+
+class DiffeoSymTensor3DVolInputSpec(CommandLineInputSpec):
+    in_file = File(desc='moving tensor volume', exists=True,
+                   argstr="-in %s", mandatory=True)
+    out_file = File(desc='output filename',
+                    argstr="-out %s", name_source="in_file",
+                    name_template="%s_diffeoxfmd", keep_extension=True)
+    transform = File(exists=True, argstr="-trans %s",
+                     mandatory=True, desc='transform to apply')
+    df = traits.Str('FD', argstr="-df %s", usedefault=True)
+    interpolation = traits.Enum('LEI', 'EI', usedefault=True,
+                                argstr="-interp %s",
+                                desc='Log Euclidean/Euclidean Interpolation')
+    reorient = traits.Enum('PPD', 'FS', argstr='-reorient %s',
+                           usedefault=True, desc='Reorientation strategy: '
+                           'preservation of principal direction or finite '
+                           'strain')
```

Comment:
Just to be clear, no reorientation is not an option for this command? And if you omitted the argument, it would either cause an error or default to `PPD`?

Reply:
I think that is the case (see, below, from documentation). It looks like the available options are specified to the right and "FS" or "PPD" are the only two options.

-reorient type: string(s) required: N	default: PPD FS or PPD

Reply:
Cool. This looks good then.